digitalocean/spaces_buckets.go

```diff
@@ -7,6 +7,11 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+var (
+	// SpacesRegions is a list of DigitalOcean regions that support Spaces.
+	SpacesRegions = []string{"ams3", "fra1", "nyc3", "sfo2", "sfo3", "sgp1"}
```

Comment:
should we make an enum file for these values? Dynamically loading from an enum file can make updating said file much easier.

Reply:
For use in a `ValidateFunc` we'd still need to return a list of strings. Feels a bit heavy to me.

```go
type SpacesRegion int

const (
	SpacesRegionAMS3 SpacesRegion = iota
	SpacesRegionFRA1
	SpacesRegionNYC3
	SpacesRegionSFO2
	SpacesRegionSFO3
	SpacesRegionSGP1
)

func (s SpacesRegion) String() string {
	return [...]string{"ams3", "fra1", "nyc3", "sfo2", "sfo3", "sgp1"}[s]
}

func SpacesRegionValues() []string {
	return []string{
		SpacesRegionAMS3.String(),
		SpacesRegionFRA1.String(),
		SpacesRegionNYC3.String(),
		SpacesRegionSFO2.String(),
		SpacesRegionSFO3.String(),
		SpacesRegionSGP1.String(),
	}
}
```

Or the slightly simplified but not really an enum approach:

```go
const (
	SpacesRegionAMS3 = "ams3"
	SpacesRegionFRA1 = "fra1"
	SpacesRegionNYC3 = "nyc3"
	SpacesRegionSFO2 = "sfo2"
	SpacesRegionSFO3 = "sfo3"
	SpacesRegionSGP1 = "sgp1"
)

func SpacesRegionValues() []string {
	return []string{
		SpacesRegionAMS3,
		SpacesRegionFRA1,
		SpacesRegionNYC3,
		SpacesRegionSFO2,
		SpacesRegionSFO3,
		SpacesRegionSGP1,
	}
}
```

If we were building out a library for working with Spaces (i.e. adding Spaces support to godo) it might be worth it. Not so sure as an internal convenience. Maybe it just shouldn't be exported?

Reply:
both of these seem to much work for a simple slice that's used only once of twice. LGTM